### PR TITLE
[hotfix][e2e] refactor jupiter,add junit-jupiter-api

### PR DESCRIPTION
**Pull request by liugddx**
## Purpose of this pull request

refactor jupiter,add junit-jupiter-api
## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by TyrantLucifer** [comment]
Can you explain why you want to add this dependency?

**Comment by CalvinKirs** [comment]
the e2e module used junit 4.

**Comment by liugddx** [comment]
Occasionally switching branches leads to inconsistent versions of api dependencies
ecfb805b4e601305b5f182eee5db365

**Review by CalvinKirs — COMMENTED** [review]
I think we have declared these dependencies, `junit-jupiter-params` inculde `juitper-api`
if you checked. e2e We use junit4 and 5 for `Testcontainers` reasons

**Comment by liugddx** [comment]
> I think we have declared these dependencies, `junit-jupiter-params` inculde `juitper-api` if you checked. e2e We use junit4 and 5 for `Testcontainers` reasons

but the e2e dependence `jupiter`.if the `junit-jupiter-api` has a different versions.the error will happen.

File: pom.xml [hunk 1]
```diff
@@ -256,13 +256,20 @@
...
                 <artifactId>jcommander</artifactId>
                 <version>${jcommander.version}</version>
             </dependency>
 
+            <dependency>
+                <groupId>org.junit.jupiter</groupId>
+                <artifactId>junit-jupiter-api</artifactId>
+                <version>${junit.version}</version>
+                <scope>test</scope>
+            </dependency>
```
Comment (ashulin) [inline pom.xml]: `junit-jupiter-engine` contains `junit-jupiter-api`.

**Comment by ashulin** [comment]
image
In which dependency will `junit-jupiter-api-5.8.2`  be referenced?

File: pom.xml (hunk 1 above)
Comment (liugddx) [reply pom.xml]: It's probably because I switched branches，but strong dependence is not better?

**Comment by liugddx** [comment]
> image In which dependency will `junit-jupiter-api-5.8.2` be referenced?

i dependence `spring-boot-dependencies`,and the `junit-jupiter.version`is 5.8.2.

**Comment by ashulin** [comment]
This PR contains the reason: https://github.com/apache/incubator-seatunnel/pull/2672 . I ran CI again.

**Comment by CalvinKirs** [comment]
hi, we better revert https://github.com/apache/incubator-seatunnel/blob/34ccfd7a2cd93d8dc0d3f6bb1e1afebcc9761a9e/pom.xml#L360
This modification changes so many dependencies that I suspect there are still a lot of potential issues that haven't been discovered, but as long as we revert it, this issue and other issues can be resolved.

**Comment by CalvinKirs** [comment]
image
revert this change

**Comment by ashulin** [comment]
You need to revert the dependency changes in #2506

**Review by Hisoka-X — DISMISSED** [review]

**Comment by zhuangchong** [comment]
My suggestion is to introduce junit-bom for dependency management and remove `junit-jupiter-engine` and `junit-jupiter-params` with the following changes:

add:

```
  <dependency>
      <groupId>org.junit</groupId>
      <artifactId>junit-bom</artifactId>
      <version>${junit.version}</version>
      <type>pom</type>
      <scope>import</scope>
  </dependency>
```
remote:
```
<dependency>
      <groupId>org.junit.jupiter</groupId>
      <artifactId>junit-jupiter-engine</artifactId>
      <version>${junit.version}</version>
      <scope>test</scope>
  </dependency>

  <dependency>
      <groupId>org.junit.jupiter</groupId>
      <artifactId>junit-jupiter-params</artifactId>
      <version>${junit.version}</version>
      <scope>test</scope>
  </dependency>
```
update:
```
<dependencies>
...
 <dependency>
            <groupId>org.junit.jupiter</groupId>
            <artifactId>junit-jupiter-engine</artifactId>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.junit.jupiter</groupId>
            <artifactId>junit-jupiter-params</artifactId>
            <scope>test</scope>
        </dependency>

    </dependencies>
```

**Comment by zhuangchong** [comment]
UserControllerTest executes an error after the scheduler function is added to the web module, because the web module needs to connect to the dolphinscheduler server, I will deal with this problem.

**Comment by ashulin** [comment]
I checked the `junit-bom`, which contains only junit dependencies.

File: pom.xml [hunk 2]
```diff
@@ -178,7 +178,7 @@
         <skipUT>false</skipUT>
         <skipIT>true</skipIT>
         <elasticsearch>7</elasticsearch>
-        <slf4j.version>1.7.25</slf4j.version>
+        <slf4j.version>1.7.36</slf4j.version>
```
Comment (ashulin) [inline pom.xml]: slf4j-log4j12-1.7.36 will not include log4j.
```suggestion
        <slf4j.version>1.7.25</slf4j.version>
```

**Review by CalvinKirs — APPROVED** [review]

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**